pkg/provisioner/resourcehandlers.go

```diff
@@ -131,6 +131,53 @@ func createObjectBucket(ob *v1alpha1.ObjectBucket, c versioned.Interface, retryI
 	return
 }
 
+func updateObjectBucket(ob *v1alpha1.ObjectBucket, c versioned.Interface, retryInterval, retryTimeout time.Duration) (result *v1alpha1.ObjectBucket, err error) {
+	logD.Info("updating ObjectBucket", "name", ob.Name)
+	result, err = c.ObjectbucketV1alpha1().ObjectBuckets().Update(context.TODO(), ob, metav1.UpdateOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return result, err
+}
+
+func getObjectBucketFromClaimKey(key string, obc *v1alpha1.ObjectBucketClaim, c versioned.Interface, retryInterval, retryTimeout time.Duration) (ob *v1alpha1.ObjectBucket, err error) {
+	obName, err := objectBucketNameFromClaimKey(key)
+	if err != nil {
+		return nil, err
+	}
+
+	logD.Info("seeing if object bucket for claim exists", "expected object bucket name", obName)
+	ob, err = c.ObjectbucketV1alpha1().ObjectBuckets().Get(context.TODO(), obName, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Not found is a valid exit here which returns no error and a nil result
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	emptyRef := corev1.ObjectReference{}
+	if ob.Spec.ClaimRef == nil || *ob.Spec.ClaimRef == emptyRef {
+		// If the CRD validation for the OB improperly specifies the object reference field, it will
+		// be missing. Do our best to recover from this case gracefully.
+		logD.Info("found OB matching this OBC that does not have an object reference, assuming it should belong to this claim", "OB", ob.Name)
```

Comment:
"object reference" seems vague to me. How about:
`logD.Info("found OB matching this OBC name but the OB's ClaimRef is empty. Binding OB", ob.Name, "to OBC", obc.Name)`

Reply:
Fixing without %s/%s.`logD` automatically prints the key which identifies the OBC namespace/name.